### ISSUE TITLE
Add patches for GCC 5.4.0

### ISSUE THREAD

**Issue by piranna** [GregorR/musl-cross#65]
Add patches for GCC 5.4.0 since it's the latest one available for the 5.x branch. Also, add the patches to allow to compile it with GCC 6.x.x and newer.

**Comment by rofl0r** [comment]
this sounds like the commit message of a pull request. but i can't see any patches attached here...
did you see the comment about "project scope" in the README ?

**Comment by piranna** [comment]
Yes, I've read it, but in fact the other repo is too much difficult to understand how to integrate it. Seems the other repo generates a full cross-toolchain, that's not a bad thing, but I'm using my own one (nodeos-cross-toolchain) so I'm only interested on the GCC patch and no more. Current patches are easy to apply but the ones on the other repo it's not clear about how to apply them, so since it's from the same 5.x branch, I think it would make sense to also add an upgrade here, specially having such an important bug like not being able to compile GCC with 6.x and newer ones, so I think at least the fix patch should be applied and backported here...

**Comment by rofl0r** [comment]
i agree that it'd be nice to have but it's work...
as a quicky, you could try to just `cat *.diff > cumulative.patch` in the musl-cross-make repo in the patches/gcc-6.3.0 dir. and then add it to your toolchain or this repo here.
